Siva- Creating front-end for activity/description

pavanputti opened the pull request:
# Description
<img width="503" height="201" alt="Screenshot 2026-01-31 at 11 18 16 PM" src="https://github.com/user-attachments/assets/46225831-f861-4eef-b393-ecb634dcc1f8" />

## Related PRS (if any):
This frontend PR is related to the development backend PR. To test this frontend PR, you need to checkout the development backend PR.
…

## Main changes explained:
- Delete file A for removing unused components …
- Update file B for including new pattern …
- Create file C for introducing new components …
…

## How to test:
1. Check out the current branch.
2. Run npm install and npm run start:local (or npm start) to run this PR locally.
3. Clear site data/cache.
4. Log in as an admin user.
5. Go to Community Portal → Activity (or open http://localhost:5173/communityportal/activity/<event-id> with an event ID from GET /api/events).
6. Confirm the Activity Agenda / summary page loads and shows event title, date, type, location, status, attendance, description, and schedule (and that loading/error states do not show “Something went wrong”).

## Screenshots or videos of changes:
<img width="1439" height="858" alt="Screenshot 2026-01-31 at 11 16 51 PM" src="https://github.com/user-attachments/assets/f6bb2d27-d374-4a7f-aace-ca3d154bc805" />

pavanputti committed:
Update Activity Agenda route to support dynamic activityId parameter

- Changed route from /communityportal/ActivityAgenda to /communityportal/activity/:activityid
- Enables dynamic routing based on activity ID
- Fixes 'Page Not Found' issue when accessing activity details

pavanputti committed:
Revert "Update Activity Agenda route to support dynamic activityId parameter"

This reverts commit 32315090209d73bb82eb130b967bc436306a91f6.

pavanputti committed:
Update Activity Agenda route to support dynamic activityId parameter

- Changed route from /communityportal/ActivityAgenda to /communityportal/activity/:activityid
- Enables dynamic routing based on activity ID
- Fixes 'Page Not Found' issue when accessing activity details
- Route now accepts activityid as URL parameter

pavanputti committed:
Implement useParams hook to extract activityId from URL

- Added useParams import from react-router-dom
- Extract activityid parameter from URL route
- Added validation and error handling for missing activityId
- Component now receives activityId from URL parameter
- Prepared component for dynamic data fetching in next commit

pavanputti committed:
Implement API data fetching logic for Activity Agenda

- Added EVENT_BY_ID endpoint to URL.js for fetching single event
- Implemented useState hooks for eventData, loading, and error states
- Added useEffect to fetch event data when component mounts or activityid changes
- Implemented fallback logic: tries EVENT_BY_ID endpoint first, falls back to fetching all events and filtering by ID
- Added data transformation to map event model fields to component structure
- Implemented loading, error, and no-data states with user-friendly messages
- Added proper error handling and validation for API responses

pavanputti committed:
Refactor component to use dynamic data and enhance UI

- Enhanced data transformation with better date/time formatting helpers
- Added event metadata display (date, type, location, status, attendance)
- Improved schedule display with better formatting and resource location
- Added image error handling with fallback to default image
- Enhanced UI structure with organized sections for metadata, description, and schedule
- Improved styling with conditional dark mode support for metadata sections
- Better handling of missing or incomplete event data
- Added formatted date display with full date string
- Enhanced schedule items with background styling and better layout

pavanputti committed:
Improve error handling and fix edge cases for Activity Agenda

- Enhanced error handling with specific messages for different error types (404, network, server, permission errors)
- Added validation for activity ID (empty string check)
- Improved date/time formatting with NaN validation to prevent invalid date displays
- Enhanced schedule building with null/undefined resource filtering
- Added type checking for numeric values (maxAttendees, currentAttendees)
- Improved fallback API call error handling with network error detection
- Better event data validation (object type check)
- Enhanced error messages for better user experience
- Fixed date display logic to handle null dates properly
- Added array validation for schedule and resources data

pavanputti committed:
Refactor ActivityAgenda to reduce cognitive complexity - Extract helper functions to reduce fetchEventData complexity from 31 to ≤15 - Extract render functions to reduce ActivityAgenda complexity from 17 to ≤15

pavanputti committed:
Fix SonarQube issues: Replace isNaN with Number.isNaN and use optional chaining

pavanputti committed:
Remove unused Resources import from routes.jsx

pavanputti committed:
Merge origin/development: Resolve conflicts in routes.jsx - Keep ResourcesUsage import and ActivityAgenda route from both branches

pavanputti committed:
Merge development into siva/activity-agenda-dynamic-routing

pavanputti committed:
Activity Agenda: final polish, dark mode verification, code cleanup

- Fix dark mode by using CSS module classes (activityAgendaPage, activityAgendaDarkMode) for wrapper so theme variables apply
- Move metadata, description, and schedule inline styles into ActivityAgenda.module.css
- Add activityMetadata, metaRow, sectionHeading, descriptionBlock, scheduleItem classes with dark mode variants
- Ensure section headings use var(--heading-color); schedule items use theme-aware backgrounds
- Code cleanup: remove getPageClassName helper in favor of pageClassName; satisfy ESLint line length

pavanputti committed:
Fix ActivityAgenda: use pageClassName in renderLoading and renderNoData

Resolves ReferenceError 'getPageClassName is not defined' when loading or showing no-data state.

pavanputti committed:
Merge remote-tracking branch 'origin/development' into siva/activity-agenda-dynamic-routing

HemanthNidamanuru reviewed (dismissed):
Hi,

I tested this PR locally and the Activity Agenda page loads correctly. The event title, date, type, location, status, attendance, description, and schedule are all displayed as expected, and I did not see any error messages.

I noticed some misalignment in the elements on the page. Is the alignment also part of this PR?

<img width="1903" height="1035" alt="Image" src="https://github.com/user-attachments/assets/4fd141b7-816a-4e6f-bf59-e04ff8508de1" />
<img width="1896" height="1038" alt="Image" src="https://github.com/user-attachments/assets/42c1ba7e-283e-4916-a90b-958442a752ed" />

Anusha-Gali reviewed (changes requested):
Hi Siva,

I have reviewed your PR locally and although it works as per requirement for the very first project id in both the modes, in all the other projects the image size is very large and is messing up the alignment of the text fields.

<img width="1279" height="759" alt="Image" src="https://github.com/user-attachments/assets/f15c7b55-603a-4907-bab9-31134b5f4d67" />
<img width="1279" height="759" alt="Image" src="https://github.com/user-attachments/assets/3a3c9e08-9e0d-419d-9380-db5cdc104b99" />

<img width="1279" height="759" alt="Image" src="https://github.com/user-attachments/assets/8d832a3c-41ca-4786-9a83-f27e22ca03b4" />


Hi @Anusha-Gali ,

Thanks for pointing this out. I’ll fix the image sizing for the other project IDs so it doesn’t affect the text alignment. I’ll update it in PR #4434: https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/pull/4434
. Thanks!

pavanputti committed:
Fix Activity Agenda image size and layout alignment

- Constrain image column (max-width 380px) and image (max-height 320px, object-fit cover)
- Add flex gap and align-items for consistent layout across events
- Prevent large cover images from breaking text alignment in light/dark mode
- Responsive: image max-height 280px on mobile, content full width

pavanputti committed:
Fix SonarQube issues: remove unused imports and refactor nested template literals

- routes.jsx: Remove unused PRDashboardOverview, PRDashboardPromotionEligibility, PRDashboardTopReviewedPRs, PRDashboardDetails imports; remove commented PurchaseEquipment code
- URL.js: Refactor TOP_CONVERTED and LEAST_CONVERTED to avoid nested template literals

pavanputti committed:
Merge development into siva/activity-agenda-dynamic-routing; resolve URL.js conflict

Keep SonarQube-refactored TOP_CONVERTED/LEAST_CONVERTED (no nested template literals).

pavanputti committed:
Fix failing tests: use PRDashboard for missing PR dashboard routes

PRDashboardOverview, PRDashboardPromotionEligibility, PRDashboardTopReviewedPRs,
and PRDashboardDetails components do not exist; use existing PRDashboard for
/pr-dashboard/overview, promotion-eligibility, top-reviewed-prs, and details routes.

rohanrastogi311 reviewed (dismissed):
Hi Siva, 

Well done with this implementation. 

<img width="1728" height="1042" alt="Image" src="https://github.com/user-attachments/assets/a4236c12-538f-45ed-9125-9683e3baf775" />

Anusha-Gali reviewed (dismissed):
Hi Siva,

Thanks for fixing the issue. The UI is as per requirement.

<img width="1274" height="829" alt="Image" src="https://github.com/user-attachments/assets/f499d21e-1c99-4b0f-8d9d-a8791b7d1375" />
<img width="1274" height="829" alt="Image" src="https://github.com/user-attachments/assets/ab88dfc1-8312-4764-a406-0d577cc8dab3" />

pavanputti committed:
Merge remote-tracking branch 'origin/development' into siva/activity-agenda-dynamic-routing

# Conflicts:
#	src/routes.jsx

one-community commented:
Thank you all, merging!
